WireAPI/Sources/WireAPI/APIs/UserClientsAPI/UserClientsAPIV7.swift

```diff
@@ -16,6 +16,91 @@
 // along with this program. If not, see http://www.gnu.org/licenses/.
 //
 
+import Foundation
+
 final class UserClientsAPIV7: UserClientsAPIV6 {
+
     override var apiVersion: APIVersion { .v7 }
+
+    override func getSelfClients() async throws -> [SelfUserClient] {
+        let components = URLComponents(string: "\(pathPrefix)/clients")
+
+        guard let url = components?.url else {
+            assertionFailure("generated an invalid url")
+            throw UserClientsAPIError.invalidURL
+        }
+
+        let request = URLRequestBuilder(url: url)
+            .withMethod(.get)
+            .build()
+
+        let (data, response) = try await apiService.executeRequest(
+            request,
+            requiringAccessToken: true
+        )
+
+        return try ResponseParser()
+            .success(code: .ok, type: ListUserClientV7.self)
+            .parse(code: response.statusCode, data: data)
+    }
+}
+
+struct ListUserClientV7: Decodable, ToAPIModelConvertible {
+
+    let payload: [SelfUserClientV7]
+
+    init(from decoder: any Decoder) throws {
+        let container = try decoder.singleValueContainer()
+        let payload = try container.decode([SelfUserClientV7].self)
+        self.payload = payload
+    }
+
+    func toAPIModel() -> [SelfUserClient] {
+        payload.map { $0.toAPIModel() }
+    }
+}
+
+struct SelfUserClientV7: Decodable, ToAPIModelConvertible {
+
+    let id: String
+    let type: UserClientType
+    let activationDate: UTCTimeMillis
+    let label: String?
+    let model: String?
+    let deviceClass: DeviceClass?
+    let lastActiveDate: UTCTime?
+    let mlsPublicKeys: MLSPublicKeys?
+    let cookie: String?
+    let capabilities: [UserClientCapability]?
```

Comment:
@caldrian  I would add a small comment to specify what changed in v7 (removal of the object container)

Reply:
fixed, thank you for the hint!